server: remove interfaces with single implementation

stevendanna opened the pull request:
This removes some interfaces with only 1 implementation from the server controller code and moves more of the server controller methods into server_controller.go

While the intent of these interfaces was to facilitate testing, that work was never completed and the currently they mostly serve to make the code a bit harder to navigate.

While having the code in shorter files is sometimes nice, in this case I think in this case it is mostly a hinderance to seeing the full scope of behaviors.

Epic: none
Release note: None

stevendanna committed:
server: remove interfaces with single implementation

This removes some interfaces with only 1 implementation from the
server controller code and moves more of the server controller methods
into server_controller.go

While the intent of these interfaces was to facilitate testing, that
work was never completed and the currently they mostly serve to make
the code a bit harder to navigate.

While having the code in shorter files is sometimes nice, in this case
I think in this case it is mostly a hinderance to seeing the full
scope of behaviors.

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/120207)
<!-- Reviewable:end -->

stevendanna commented:
For your consideration as you start addressing issues in this code. I think de-interfacing some of the code will make it a bit easier to iterate on improving it.

dt commented:
Thanks!

bors r+
